virttest/utils_disk.py

```diff
@@ -600,6 +602,154 @@ def create_filesystem(session, partition_name, fstype, ostype, timeout=360):
         create_filesyetem_linux(session, partition_name, fstype, timeout)
 
 
+def get_fstype_mpoint_linux(session, partition):
+    """
+    Get a partition's file system type and mountpoint.
+
+    :param session: session object to guest.
+    :param partition: disk partition, like /dev/sdb1.
+    :return: Tuple (fstype, mountpoint)
+    """
+    mount_list = session.cmd_output('cat /proc/mounts')
+    pattern = r'%s\s(.+?)\s(.+?)\s' % partition
+    mount_info = re.search(pattern, mount_list)
+    if mount_info:
+        mountpoint = mount_info.group(1)
+        fstype = mount_info.group(2)
+        return (fstype, mountpoint)
+    logging.error('partition is not mounted, mounted'
+                  'list as:\n%s' % mount_list)
+
+
+def resize_filesystem_linux(session, partition, size):
+    """
+    resize file system in linux guest.
+    support enlarge and shrink.
+
+    :param session: session object to guest.
+    :param partition: disk partition, like /dev/sdb1.
+    :param size: resize file system to size.
+                 size unit can be 'B', 'M', 'G'.
+                 support transfer size with AVAILABLE_SIZE,
+                 enlarge to maximun available size.
+    """
+    def resize_xfs_fs(size):
+        if size == AVAILABLE_SIZE:
+            resize_fs_cmd = 'xfs_growfs -d %s' % mountpoint
+        else:
+            size = utils_numeric.normalize_data_size(size, order_magnitude="B")
+            get_bsize_cmd = 'xfs_growfs -n %s' % mountpoint
+            output = session.cmd_output(get_bsize_cmd)
+            bsize = re.findall(r'data\s+=\s+bsize=(\d+)', output, re.M)[0]
+            blocks = int(size.split(".")[0]) // int(bsize) - 512
```

Comment:
How do you confirm to need to subtract 512 blocks in `blocks = int(size.split(".")[0]) // int(bsize) - 512` , not 1024 blocks or 4096 blocks. Is it related with the options "physical_block_size" and "logical_block_size" ?
e.g:
```
-drive id=drive_disk1,if=none,snapshot=off,aio=native,cache=none,format=qcow2,file=/root/test_home/yhong/images/storage.qcow2 \
-device scsi-hd,id=disk1,drive=drive_disk1,physical_block_size=4096,logical_block_size=1024 \
```

Reply:
Has updated, thanks~